modules/sw360/src/test/java/org/eclipse/sw360/antenna/workflow/generator/SW360UpdaterTest.java

```diff
@@ -129,4 +127,47 @@ public String getLinkStr() {
         return artifact;
     }
 
+    @Test
+    public void testMainLicenseIsEmptyAndFinalLicenseNull() {
+        Artifact artifact  = new Artifact("JSON");
+        artifact.setProprietary(false);
+        artifact.addFact(new MavenCoordinates("artifactId(test)", "org.group.id", "1.2.3"));
+        artifact.addFact(new DeclaredLicenseInformation(new LicenseInformation() {
```

Comment:
Why is here a custom inline `LicenseInformation` class used? That might implement behaviour which the actual implementations can not reproduce. Maybe test with actual `License` or `LicenseStatement`.

Reply:
Since in the Updater the `getLicenses` and `evaluate` functions of the Licenseinformation are called to get the information transferred to sw360, this seemed to me like the most reasonable approach to ensure that what we wanted to reach (empty License in Main License and hence empty string in evaluate are not mapped to the release) is accomplished. 

Using an actual License or LicenseStatement would not make that condition testable, I believe. Could be wrong though, open for suggestions.

Reply:
> Using an actual License or LicenseStatement would not make that condition testable, I believe. Could be wrong though, open for suggestions.

but if it is not a reachable state with our implementation, the test is also useless?

Reply:
No, I believe that also if we don'T have any license i think evaluate return "", so this should still be tested

Reply:
but than it can be simulated by
```
artifact.addFact(new DeclaredLicenseInformation(null));
```
or
```
artifact.addFact(new DeclaredLicenseInformation(new LicenseStatement()));
```
?

Reply:
I tested and new `LicenseStatement` when put as `ArtifactFact` results to true when:
 `assertThat(ArtifactLicenseUtils.getFinalLicenses(artifact).evaluate()).isEqualTo("");`
And tests still pass

So, I will change it as you suggested. Thanks